File: AuctionAdmin/src/main/webapp/js/core/site.js
/* 
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

var site = (function(){
    
    var baseUri = "http://localhost:8080/AuctionAdmin/";
    var auctionobjects = new AuctionCatalogue(baseUri + "auctionObjects");
    var customers = new CustomerCatalogue(baseUri + "customers");
    
    return {
        getAuctionCatalogue : function() {
            return auctionobjects;
        },
        getCustomerCatalogue : function() {
            return customers;
        },
        getBaseUri : function(){
            return baseUri;
        }
    };
})();

File: AuctionAdmin/src/main/java/admin/src/Site.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package admin.src;

import longcat.auction.src.AuctionCatalogue;
import longcat.auction.src.CustomerCatalogue;
import longcat.auction.src.SuperSite;

/**
 * Connection with the model
 * @author Alexander Lissenko
 */
public enum Site {
    INSTANCE;
    
    private final SuperSite site;
    
    private Site(){
        site = new SuperSite();
    }
    
    public AuctionCatalogue getAuctionCatalogue() {
        return site.getAuctionCatalogue();
    }
    
    public CustomerCatalogue getCustomerCatalogue() {
        return site.getCustomerCatalogue();
    }
}
